tests/blackbox_test.go

```diff
@@ -0,0 +1,1215 @@
+	}
+}
+
+func intJoin(ints []int, delimiter string) string {
+	strArr := []string{}
+	for _, i := range ints {
+		strArr = append(strArr, strconv.Itoa(i))
+	}
+	return strings.Join(strArr, delimiter)
+}
+
+func removeEmpty(strings []string) []string {
+	var r []string
+	for _, str := range strings {
+		if str != "" {
+			r = append(r, str)
+		}
+	}
+	return r
+}
+
+func generateUUID(t *testing.T) string {
+	out, err := exec.Command("/usr/bin/uuidgen").CombinedOutput()
+	if err != nil {
+		t.Fatal("uuidgen", err)
+	}
+	return strings.TrimSpace(string(out))
+}
+
+func createFiles(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.Files == nil {
+			continue
+		}
+		for _, file := range partition.Files {
+			err := os.MkdirAll(filepath.Join(
+				partition.MountPath, file.Path), 0644)
+			if err != nil {
+				t.Fatal("mkdirall", err)
+			}
+			f, err := os.Create(filepath.Join(
+				partition.MountPath, file.Path, file.Name))
+			defer f.Close()
+			if err != nil {
+				t.Fatal("create", err, f)
+			}
+			if file.Contents != nil {
+				writer := bufio.NewWriter(f)
+				writeStringOut, err := writer.WriteString(
+					strings.Join(file.Contents, "\n"))
+				if err != nil {
+					t.Fatal("writeString", err, string(writeStringOut))
+				}
+				writer.Flush()
+			}
+		}
+	}
+}
+
+func unmountRootPartition(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.Label != "ROOT" {
+			continue
+		}
+
+		umountOut, err := exec.Command(
+			"/bin/umount", partition.Device).CombinedOutput()
+		if err != nil {
+			t.Fatal("umount", err, string(umountOut))
+		}
+	}
+}
+
+func unmountPartitions(t *testing.T, partitions []*Partition) {
+	for _, partition := range partitions {
+		if partition.FilesystemType == "" || partition.Label == "ROOT" {
+			continue
+		}
+		umountOut, err := exec.Command(
+			"/bin/umount", partition.Device).CombinedOutput()
+		if err != nil {
+			t.Fatal("umount", err, string(umountOut))
+		}
+	}
+}
+
+func setExpectedPartitionsDrive(actual []*Partition, expected []*Partition) {
+	for _, a := range actual {
+		for _, e := range expected {
+			if a.Number == e.Number {
+				e.MountPath = a.MountPath
+				e.Device = a.Device
+				break
+			}
+		}
+	}
+}
+
+func validatePartitions(
+	t *testing.T, expected []*Partition, fileName string) {
+	for _, e := range expected {
+		if e.TypeCode == "blank" {
+			continue
+		}
+		sgdiskInfo, err := exec.Command(
+			"/sbin/sgdisk", "-i", strconv.Itoa(e.Number),
+			fileName).CombinedOutput()
+		if err != nil {
+			t.Fatal("sgdisk -i", strconv.Itoa(e.Number), err)
+		}
+		lines := strings.Split(string(sgdiskInfo), "\n")
+		actualTypeGUID := strings.ToUpper(strings.TrimSpace(
+			strings.Split(strings.Split(lines[0], ": ")[1], " ")[0]))
+		actualSectors := strings.Split(strings.Split(lines[4], ": ")[1], " ")[0]
+		actualLabel := strings.Split(strings.Split(lines[6], ": ")[1], "'")[1]
```

Comment:
Same comment as earlier that this will panic if the output isn't what you expect, and that it might be fine.